src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -42,54 +39,68 @@
+     * @deprecated use {@link #StandardSlackService(String, String, boolean, String, boolean, String, String, String)} instead}
      */
     @Deprecated
     public StandardSlackService(String baseUrl, String teamDomain, String token, String authTokenCredentialId, boolean botUser, String roomId) {
-        this(baseUrl, teamDomain, token, authTokenCredentialId, botUser, roomId, false);
+        this(baseUrl, teamDomain, token, authTokenCredentialId, botUser, roomId, false, null, null);
     }
 
     /**
-     * @deprecated use {@link #StandardSlackService(String, String, boolean, String, boolean, String)} instead}
+     * @deprecated use {@link #StandardSlackService(String, String, boolean, String, boolean, String, String, String)} instead}
      */
     @Deprecated
-    public StandardSlackService(String baseUrl, String teamDomain, String token, String authTokenCredentialId, boolean botUser, String roomId, boolean replyBroadcast) {
-        this(baseUrl, teamDomain, botUser, roomId, replyBroadcast);
+    public StandardSlackService(String baseUrl, String teamDomain, String token, String authTokenCredentialId, boolean botUser, String roomId, boolean replyBroadcast, String iconEmoji, String username) {
+        this(baseUrl, teamDomain, botUser, roomId, replyBroadcast, iconEmoji, username);
         this.populatedToken = getTokenToUse(authTokenCredentialId, token);
         if (this.populatedToken == null) {
             throw new IllegalArgumentException("No slack token found, setup a secret text credential and configure it to be used");
         }
     }
 
+
+    @Deprecated
+    public StandardSlackService(String baseUrl, String teamDomain, boolean botUser, String roomId, boolean replyBroadcast, String populatedToken) {
+        this(baseUrl, teamDomain, botUser, roomId, replyBroadcast, null, null);
+        if (populatedToken == null) {
+            throw new IllegalArgumentException("No slack token found, setup a secret text credential and configure it to be used");
+        }
+        this.populatedToken = populatedToken;
+    }
+
     /**
      * @param baseUrl        the full url to use, this is an alternative to specifying teamDomain
      * @param teamDomain     the teamDomain inside slack.com to use
      * @param botUser
      * @param roomId         a semicolon separated list of rooms to notify
      * @param replyBroadcast
+     * @param iconEmoji
+     * @param username
      * @param populatedToken a non-null token to use for authentication
      */
-    public StandardSlackService(String baseUrl, String teamDomain, boolean botUser, String roomId, boolean replyBroadcast, String populatedToken) {
-        this(baseUrl, teamDomain, botUser, roomId, replyBroadcast);
+    public StandardSlackService(String baseUrl, String teamDomain, boolean botUser, String roomId, boolean replyBroadcast, String iconEmoji, String username, String populatedToken) {
```

Comment:
this also needs a builder...

Reply:
Now that I have a working example, I should be able to do a better job setting one up.